src/System.Private.ServiceModel/src/System/ServiceModel/Channels/HttpChannelFactory.cs

```diff
@@ -697,6 +838,31 @@ private void TryCompleteHttpRequest(HttpRequestMessage request)
                         _channel.OnHttpRequestCompleted(request);
                     }
                 }
+
+                private async Task SendPreauthenticationHeadRequestIfNeeded()
+                {
+                    if (_factory.AuthenticationScheme == AuthenticationSchemes.Anonymous)
+                    {
+                        return;
+                    }
```

Comment:
NIT: Above could be replaced with 

    if (!AuthenticationSchemeMayRequireResend)
    {
        return; 
    }